PokeDemo/Presenter/PokemonListPresenter.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  PokemonListPresenter.swift
+//  PokeDemo
+//
+//  Created by Rudr Bansal on 15/07/22.
+//
+
+import Foundation
+import UIKit
+
+protocol PokemonListPresenterDelegate: AnyObject {
+    func show(_ pokemons: [Pokemon])
+    func showAlert(title: String, message: String)
+}
+
+typealias PokemonListViewPresenterDelegate = PokemonListPresenterDelegate & UIViewController
+
+final class PokemonListPresenter {
+    
+    // MARK: Properties
+    
+    // MARK: Public
+    
+    weak var delegate: PokemonListViewPresenterDelegate?
+    
+    // MARK: Private
+    
+    private let service: ServiceRepresentable
+    
+    // MARK: Initailizers
+    
+    init(service: ServiceRepresentable = Service.shared) {
+        self.service = service
+    }
+    
+    // MARK: Exposed methods
+    
+    func viewDidLoad() {
+        getPokemons()
+    }
+    
+    func setViewDelegate(delegate: PokemonListViewPresenterDelegate) {
+        self.delegate = delegate
+    }
```

Comment:
delegate property is already exposed. Either make delegate property private or remove this function

Reply:
Covered in b5fd69676f638e5f3cf7d43fbc7eb55c66478266